include/c.h

```diff
@@ -82,6 +82,41 @@
 # endif
 #endif
 
+/* switch fallthrough */
+#ifdef __has_attribute
+# define C_HAS_ATTRIBUTE(x) __has_attribute(x)
+#else
+# define C_HAS_ATTRIBUTE(x) 0
+#endif
+
+/* C-language Attributes are added in C23. */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ > 201710L) && defined(__has_c_attribute)
+# define C_HAS_C_ATTRIBUTE(x) __has_c_attribute(x)
+#else
+# define C_HAS_C_ATTRIBUTE(x) 0
+#endif
+
+#if defined(__cplusplus) && defined(__has_cpp_attribute)
+# define C_HAS_CPP_ATTRIBUTE(x) __has_cpp_attribute(x)
+#else
+# define C_HAS_CPP_ATTRIBUTE(x) 0
+#endif
+
+/*
+Define FALLTHROUGH macro for annotating switch case with the 'fallthrough' attribute
+introduced in CPP17 and C23.
+CPP17 : https://en.cppreference.com/w/cpp/language/attributes/fallthrough
+C23   : https://en.cppreference.com/w/c/language/attributes/fallthrough
+*/
+#if C_HAS_C_ATTRIBUTE(x)
```

Comment:
Is passing `x` to `C_HAS_C_ATTRIBTE` correct?
I think this should be
```
#if C_HAS_C_ATTRIBTE(fallthrough)
```

Reply:
Sorry. I found `C_HAS_C_ATTRIBUTE(x)` came from `xxhash.h`.
At the upstream project (), this onee is fixed: https://github.com/Cyan4973/xxHash/commit/4640547143e904a107f040e76cd9a0e1f2ed0661
https://github.com/Cyan4973/xxHash/blob/953a09abc39096da9e216b6eb0002c681cdc1199/xxhash.h#L827

What do you think about prepending a commit porting this fix to your pull request?

Reply:
will do